db/change_cache.go

```diff
@@ -36,6 +36,18 @@ const (
 	unusedSeqCollectionID         = 0                // Collection ID used by ChangeWaiter to mark unused sequences
 )
 
+// DocumentType indicates the type of document being processed in the change cache (e.g. User doc etc).
+type DocumentType uint8
+
+const (
+	DocTypeDocument       DocumentType = iota // Customer data document type
```

Comment:
```suggestion
	DocTypeUnknown       DocumentType = iota // Unknown document type
	DocTypeCustomerData                      // Customer data document type
```

Maybe this instead? So we can treat the `0` value as unknown for all non-recognized `_sync:` docs, and then have `DocTypeCustomerData` be used for everything that isn't `_sync:` prefixed.

It makes the switch/case suggested easier if there's a doc type that represents this that can no-op.